### DISCUSSION TITLE
Extract object data for bounding boxes

### DISCUSSION THREAD

**Discussion by nbsusa** [ImageMagick/ImageMagick#2213] [Help]
ImageMagick-7.0.6-Q16

I am trying to export a text file of the bounding box coordinates for a mask of multiple images on a page. Running Windows 10 command line and using -define connected-components:verbose=true. This shows me what I need as far as the coordinates but how do I get just that info (and maybe the file name) into a text file? In the CMD window it shows OBJECTS (ID: Bounding Box, etc.) as well as the various coordinates for the multiple images in my mask.

I've tried many things (ie; -write, identify, etc.) but just can't get it to write any more than the original image data.

**Comment by nbsusa** [comment]
Here is the code without any of the "write" parts I tried:  

convert  "INPUT.tif" -threshold 30%% -type bilevel -define connected-components:verbose=true -define connected-components:area-threshold=5000 -define connected-components:mean-color=true -connected-components 4 -morphology smooth octagon:5 -transparent black -compress zip  "OUTPUT.tif"

**Comment by fmw42** [comment]
You will need pipe the textual output from the command and script it to extract just the bounding boxes. Sorry, I can do that in Unix syntax, but do not know the equivalent in Windows. One of the Windows users will need to help on that.

**Reply by nbsusa** [reply]
Thanks Fred! I'll see if I can figure out how to do that and if not hopefully a windows user will reply.

**Comment by snibgo** [comment]
```
( for /F "usebackq skip=1 tokens=2-5 delims=:x+ " %%A in (`magick ^
  in.png ^
  -define "connected-components:verbose=true" ^
  -connected-components 4 ^
  NULL:`) do @echo %%A,%%B,%%C,%%D
) >out.txt
```
You can put all your other options in the "magick" command, if you want.

**Reply by nbsusa** [reply]
Incredible! Thank you very much.
